landlab/components/concentration_tracker/concentration_tracker_for_diffusion.py

```diff
@@ -0,0 +1,390 @@
+        concentration_in_bedrock=0,
+        concentration_from_weathering=None,
+    ):
+        """
+        Parameters
+        ----------
+        grid: ModelGrid
+            Landlab ModelGrid object
+        concentration_initial: float, array, or str, optional
+            Initial concentration in soil/sediment as either a scalar, array,
+            or the name of an existing node field, -/m^3.
+        concentration_in_bedrock: float, array, or str, optional
+            Concentration in bedrock as either a scalar, array, or the name
+            of an existing node field, -/m^3.
+        concentration_from_weathering: float or array, optional
+            Concentration generated during the weathering process, -/m^3.
+            Defaults to ``None``, which causes all weathered bedrock to retain its
+            original parent material concentration (``concentration_in_bedrock``)
+            as it weathers to soil. Use this parameter to differentiate between
+            the concentration in weathered material compared to its parent bedrock.
+        """
+
+        super().__init__(grid)
+        # Store grid and parameters
+
+        # use setters for C_init, C_br, and C_w defined below
+        self.C_init = concentration_initial
+        self.C_br = concentration_in_bedrock
+
+        # get reference to inputs
+        self._soil__depth = self._grid.at_node["soil__depth"]
+        self._soil__depth_old = self._soil__depth.copy()
+        self._soil_prod_rate = self._grid.at_node["soil_production__rate"]
+        self._flux = self._grid.at_link["soil__flux"]
+
+        # create outputs if necessary and get reference.
+        self.initialize_output_fields()
+
+        # Define concentration field (if all zeros, then add C_init)
+        if np.allclose(self._grid.at_node["sediment_property__concentration"], 0.0):
+            self._grid.at_node["sediment_property__concentration"] += self.C_init
+        self._concentration = self._grid.at_node["sediment_property__concentration"]
+
+        if np.allclose(self._grid.at_node["bedrock_property__concentration"], 0.0):
+            self._grid.at_node["bedrock_property__concentration"] += self.C_br
+        self.C_br = self._grid.at_node["bedrock_property__concentration"]
+
+        # use setter for C_w defined below
+        self.C_w = concentration_from_weathering
+
+        # Sediment property concentration field (at links, to calculate dQCdx)
+        self._C_links = np.zeros(self._grid.number_of_links)
+
+        # Sediment property mass field (at links, to calculate dQCdx)
+        self._QC_links = np.zeros(self._grid.number_of_links)
+
+    @property
+    def C_init(self):
+        """Initial concentration in soil/sediment (kg/m^3)."""
+        return self._C_init
+
+    @property
+    def C_br(self):
+        """Concentration in bedrock (kg/m^3)."""
+        return self._C_br
+
+    @property
+    def C_w(self):
+        """Concentration from the weathering process (kg/m^3)."""
```

Comment:
What do you think about changing the names of these properties to something more descriptive?

Maybe `conc_init`, `conc_bedrock`, `conc_weathering`? Or maybe `concentration_` instead of `conc_` but maybe that's too long?

Reply:
I've gone for conc_ for brevity.